### PR TITLE
PR: Add an menu entry to select a file name and insert it in the Editor

### LINKED ISSUE #2641: Enhancement: insert filename as compatible path
Enhancement idea:

# Rationale:
When doing data analysis, I often struggle with the way path names are defiend in python & windows.

Due to spaces and backslashes in the file names the scipts often return errors because they cannot find the files.

# Suggestion:

A context-menu entry for the file explorer or file picker with:

* `insert with absolute path here`
* `insert as relative path here`

and if needed, the editor can propse the import of `os` / `sys` packages.

This would be great helper and get the filepath issues out of way.

### PR DESCRIPTION
**Pull request by chends888**
### Pull Request Checklist

* [x] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md)
* [x] Based your PR on the latest version of the correct branch (master or 3.x)
* [x] Followed [PEP8](https://www.python.org/dev/peps/pep-0008/) for code style
* [x] Ensured your pull request hasn't eliminated unrelated blank lines/spaces,
      modified the ``spyder/defaults`` directory, or added new icons/assets
* [x] Wrote at least one-line docstrings for any new functions
* [x] Added at least one unit test covering the changes, if at all possible
* [x] Described your changes and the motivation for them below
* [x] Noted what issue(s) this pull request resolves, creating one if needed
* [x] Included a screenshot, if this PR makes any visible changes to the UI

## Description of Changes

File: codeeditor.py:
-Imported function getopenfilename from the already imported _qtpy.compat_ library
-Added function _get_filepath_
-Added _Insert file path to options when right-clicking the Spyder editor_

insert_file_path_screenshot
insert_file_path_screenshot2

### Issue(s) Resolved

Fixes #2641

### CODE REVIEW

**Comment by goanpeca** [comment]
Hi @chends888 thanks for working in here. Was also part of this fixing a way to include these things from the File explorer?

**Comment by chends888** [comment]
@goanpeca Ty! And what do you mean these things?

**Comment by ccordoba12** [comment]
@chends888, thanks for your help with this one! It's really appreciated.

@goanpeca means that you should also be able to right click on a file in our File Explorer and send its path to the Editor.

**Comment by chends888** [comment]
@ccordoba12 Oh nice! I didn't think about that. Would that be better than what it is now? Cause I think it wasn't part of the original issue #2641 or was it?

**Comment by ccordoba12** [comment]
Please read the first post in that issue again:

> A context-menu entry for the file explorer or file picker with:
    insert with absolute path here
    insert as relative path here

So until now you have implemented the second suggestion, but the first one is still missing.

File: spyder/plugins/editor/widgets/codeeditor.py [hunk 1]
```diff
@@ -2813,6 +2813,9 @@ def setup_context_menu(self):
         zoom_reset_action = create_action(
             self, _("Zoom reset"), shortcut=QKeySequence("Ctrl+0"),
             triggered=self.zoom_reset.emit)
+        insert_filepath = create_action(
+            self, _("Insert file path"), shortcut=QKeySequence(""),
+            triggered=self.insert_filepath)
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/codeeditor.py]: Please also add this action to our *Source* menu.

Comment (chends888) [reply spyder/plugins/editor/widgets/codeeditor.py]: @ccordoba12 Oh nice, I'll do it right away! Thanks

Comment (chends888) [reply spyder/plugins/editor/widgets/codeeditor.py]: Shoud I remove it from the *right-click menu*? Or just add to the *source* menu?

Comment (ccordoba12) [reply spyder/plugins/editor/widgets/codeeditor.py]: Please add it to the source menu and leave it in the right-click one.

Comment (chends888) [reply spyder/plugins/editor/widgets/codeeditor.py]: Ok!

File: spyder/plugins/editor/widgets/codeeditor.py [hunk 2]
```diff
@@ -3291,6 +3294,15 @@ def visible_blocks(self):
     def is_editor(self):
         return True
 
+    @Slot()
+    def insert_filepath(self):
+        """
+        Open a window for selecting a file,
```
Comment (CAM-Gerlach) [inline spyder/plugins/editor/widgets/codeeditor.py]: Per PEP 257, the first line of a docstring should be a single complete sentence (with a period) summarizing the purpose of the function, with subsequent sentences describing it further separated from the first by a blank line.

**Comment by CAM-Gerlach** [comment]
Hey @chends888 , really looking forward to seeing this implemented. Any updates on this? Thanks!

**Comment by chends888** [comment]
Oh hi, I've been very busy with some other things, but I'll resume working on this project next week, I think, ty!

**Comment by CAM-Gerlach** [comment]
@chends888 Noice!

**Comment by chends888** [comment]
Hi there, I've just updated the feature now you can not only add a file's path through right-clicking your code, but also from the Source Menu. Hope I get lucky to have a merged PR 😄

File: spyder/plugins/editor/plugin.py [hunk 3]
```diff
@@ -2441,6 +2447,16 @@ def remove_trailing_spaces(self):
     def fix_indentation(self):
         editorstack = self.get_current_editorstack()
         editorstack.fix_indentation()
+    @Slot()
+    def insert_filepath(self):
+        """
+        Inserts the selected file's path into the current cursor position.
+
+        When selected, a File Explorer window is opened,
```
[the remark this answers is not in the document]
Comment (chends888) [reply spyder/plugins/editor/plugin.py]: Oh, nice!

**Comment by chends888** [comment]
So, I think I'll make a pause on this issue, cause I'm short on time with some other things here, I might come back in couple of weeks ok? Sorry about that... Ty all for the support!

**Comment by CAM-Gerlach** [comment]
@chends888 Sure, of course! See you then. @ccordoba12 we could consider merging this as-is (except possibly with a test or two, and then @chends888 (or I) iteratively improving it in another PR.

**Comment by CAM-Gerlach** [comment]
@chends888 Would you agree that the functionality in #8485 supersedes that in this PR? The latter implements much more closely what was requested in #2641 , inserting absolute and relative path via a context menu in File/Project Explorer, which is much faster and more streamlined than having to select the file from the system file browser and only insert the absolute path.

**Comment by ccordoba12** [comment]
Closing in favor of #8494. Thanks for your efforts @chends888 and sorry your work was not finally merged.

**Pull request closed without merging**